test_util/convertECL.cpp

```diff
@@ -293,6 +293,7 @@ int main(int argc, char **argv)
 
     std::string rootN = filename.substr(0,p);
     std::string extension = filename.substr(p,l-p);
+    std::transform(extension.begin(), extension.end(), extension.begin(), ::toupper);  
```

Comment:
While the characters we expect for extensions to be ascii, consider the advice in the Notes section of this page: https://en.cppreference.com/w/cpp/string/byte/toupper

Reply:
>  consider the advice

More generally, though perhaps not strictly applicable to our use case, is the fact that `toupper()` is generally the [wrong solution to this problem](https://devblogs.microsoft.com/oldnewthing/20241007-00/?p=110345).

Reply:
> not strictly applicable to our use case

I think this as well: we are not doing general operations, but comparing to a very small subset of ascii strings. Therefore I think toupper() *is* an appropriate solution, but as it is not addressable you have to wrap it in a lambda as suggested in the link I posted. (Although this was not very clear. Reading your linked article clarified this better for me at least!)

Reply:
I have added the required lambda function.